code/sha3/Hacl.Hash.SHA3.Simd256.fst

```diff
@@ -0,0 +1,353 @@
+module Hacl.Hash.SHA3.Simd256
+
+open FStar.HyperStack
+open FStar.HyperStack.ST
+open FStar.Mul
+
+open LowStar.Buffer
+
+open Lib.IntTypes
+open Lib.NTuple
+open Lib.Buffer
+open Lib.ByteBuffer
+open Lib.MultiBuffer
+
+open Spec.Hash.Definitions
+open Hacl.Spec.SHA3.Vec
+open Hacl.Spec.SHA3.Vec.Common
+open Hacl.Impl.SHA3.Vec
+
+module S = Spec.SHA3
+module V = Hacl.Spec.SHA3.Vec
+module ST = FStar.HyperStack.ST
+module B = LowStar.Buffer
+module M = LowStar.Modifies
+
+#reset-options "--z3rlimit 50 --max_fuel 0 --max_ifuel 0 --using_facts_from '* -FStar.Seq'"
+
+val shake128:
+  output0:buffer_t MUT uint8
```

Comment:
why not but outputByteLen first then use `lbuffer` type for the output parameters? same with input

Reply:
It's the same parameter order that used in the original implementation with same casting technique, our team asked for this order specifically for what we thought it fits C standard convention of API argument order and former SHA3 API.

Reply:
yes, it's good to remain consistent... here's another style you might want to consider to have a more "HACL-idiomatic" style, yet keep the same order of parameters:

```
inline_for_extraction noextract
let same_as x = y:size_t { x == y }
```

val shake128:
  g_outputByteLen:Ghost.erased size_t ->
  output0: lbuffer uint8 g_outputByteLen ->
  ...
  output3: lbuffer uint8 g_outputByteLen ->
  outputByteLen: same_as g_outputByteLen
  ```
  
  this might remove a bunch of casts -- up to you to decide if you think it's worth it

Reply:
It's more good-looking style for sure. I will adapt this signature with Scalar/Simd256 functions in the next update.